IntelliTect.TestTools.TestFramework.Tests/TestDataTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using Xunit;
+
+namespace IntelliTect.TestTools.TestFramework.Tests;
+
+public class TestDataTests
+{
+    [Fact]
+    public void BlockDataT1T2ThrowsExceptionOnDuplicateTypes()
+    {
+        var exception = Assert.Throws<TypeInitializationException>(() => 
+        {
+            var _ = new BlockData<int, int>(1, 2);
+        });
+        Assert.NotNull(exception.InnerException);
+        Assert.Equal(typeof(InvalidOperationException), exception.InnerException.GetType());
```

Comment:
```suggestion
        Assert.IsType<InvalidOperationException>(exception.InnerException);
```

Same comment on the other tests below as well.